library/std/src/sys/pal/uefi/helpers.rs

```diff
@@ -222,14 +222,14 @@ pub(crate) fn runtime_services() -> Option<NonNull<r_efi::efi::RuntimeServices>>
     NonNull::new(runtime_services)
 }
 
-pub(crate) struct DevicePath(NonNull<r_efi::protocols::device_path::Protocol>);
+pub(crate) struct OwnedDevicePath(NonNull<r_efi::protocols::device_path::Protocol>);
```

Comment:
Can you clarify the purpose of this rename in the commit message? Is the intent to later have a non-owning version of DevicePath?

Reply:
Yes, a non-owning version is required when working with UEFI Shell, which manages it's own mappings and provides const pointer to the internal Device Path Protocols. I will add it in the commit message in the next push.